charts/operator/helpers.go

```diff
@@ -0,0 +1,370 @@
+	lengthToTruncate := (len(s) + len(suffix)) - 63
+	if lengthToTruncate > 0 {
+		s = helmette.Trunc(lengthToTruncate, s)
+	}
+	return fmt.Sprintf("%s-%s", s, suffix)
+}
+
+// StrategicMergePatch is a half-baked implementation of Kubernetes' strategic
+// merge patch. It's closer to a merge patch with smart handling of lists
+// that's tailored to the values permitted by [PodTemplate].
+func StrategicMergePatch(overrides *corev1.PodTemplateSpec, original corev1.PodTemplateSpec) corev1.PodTemplateSpec {
+	// TODO(chrisseto): I'd like to march this towards being a more general
+	// solution but getting go & helm to work correctly is going to take some
+	// critical thinking.
+	// - Pushing everything into a single MergeTo call won't work without VERY
+	// careful handling as `merge` is quite sensitive to the inclusion of `nil`
+	// values.
+	// - Full support of SMP (e.i. directive keys) would require a custom data
+	// type or just accepting JSON/YAML strings.
+	// - Potentially some careful handling of generics and `get` could be used
+	// to make a mostly generic SMP implementation.
+	// - Or just use real SMP in go and inject static metadata into helm to
+	// have a minimal recursive solution.
+
+	if overrides.ObjectMeta.Labels != nil {
+		original.ObjectMeta.Labels = helmette.MergeTo[map[string]string](
+			overrides.ObjectMeta.Labels,
+			helmette.Default(map[string]string{}, original.ObjectMeta.Labels),
+		)
+	}
+
+	if overrides.ObjectMeta.Annotations != nil {
+		original.ObjectMeta.Annotations = helmette.MergeTo[map[string]string](
+			overrides.ObjectMeta.Annotations,
+			helmette.Default(map[string]string{}, original.ObjectMeta.Annotations),
+		)
+	}
+
+	if overrides.Spec.SecurityContext != nil {
+		original.Spec.SecurityContext = helmette.MergeTo[*corev1.PodSecurityContext](
+			overrides.Spec.SecurityContext,
+			helmette.Default(&corev1.PodSecurityContext{}, original.Spec.SecurityContext),
+		)
+	}
+
+	if !helmette.Empty(overrides.Spec.AutomountServiceAccountToken) {
+		original.Spec.AutomountServiceAccountToken = overrides.Spec.AutomountServiceAccountToken
+	}
+
+	if overrides.Spec.ImagePullSecrets != nil && len(overrides.Spec.ImagePullSecrets) > 0 {
+		original.Spec.ImagePullSecrets = overrides.Spec.ImagePullSecrets
+	}
+
+	if !helmette.Empty(overrides.Spec.ServiceAccountName) {
+		original.Spec.ServiceAccountName = overrides.Spec.ServiceAccountName
+	}
+
+	if !helmette.Empty(overrides.Spec.NodeSelector) {
+		original.Spec.NodeSelector = helmette.MergeTo[map[string]string](
+			overrides.Spec.NodeSelector,
+			helmette.Default(map[string]string{}, original.Spec.NodeSelector),
+		)
+	}
+
+	if overrides.Spec.Affinity != nil {
+		original.Spec.Affinity = helmette.MergeTo[*corev1.Affinity](
+			overrides.Spec.Affinity,
+			helmette.Default(&corev1.Affinity{}, original.Spec.Affinity),
+		)
+	}
+
+	if overrides.Spec.TopologySpreadConstraints != nil && len(overrides.Spec.TopologySpreadConstraints) > 0 {
```

Comment:
I don't think it was necessary to fully implement this when we just need to be able to override SecurityPolicy 😅 

Did you try just calling `merge` on the pod specs and then filling in fields that need special handling? I think I tried that for containers and it didn't work but I don't recall why....

Reply:
It was not reliable, that's why I copied from redpanda chart and filled all other fields.